text/0000-emit-crate-deps.md

```diff
@@ -0,0 +1,62 @@
+- Feature Name:
+- Start Date: 2016-05-20
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a compiler flag that emits crate dependencies on a best-effort basis.
+
+# Motivation
+[motivation]: #motivation
+
+When working with projects that consist of dozens or hundreds of compilation
+units, it is useful to have some kind of automatic dependency management in the
+build system. Otherwise such information has to be maintained in at least two
+locations: Once in the compilation unit itself via an `extern crate X` statement
+and once in the build system.
+
+To this end, this RFC proposes a new compiler flag which emits the names of the
+crates a compilation unit depends on to a text file.
```

Comment:
Since we might want to add more information than just the crate name, I suggest to produce a json file (or any other structured format), which also yields the attributes given to the `extern crate` item. This is important for things like macro imports and compiler plugins.